### ISSUE TITLE
Unable to create Elasticache with multiple shards and replication group

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#691]
_This issue was originally opened by @pratincola as hashicorp/terraform#13607. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi there,

Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.

### Terraform Version
Terraform v0.9.1

### Affected Resource(s)
- aws elasticache - redis

### Terraform Configuration Files
None

### Debug Output
None

### Panic Output
None

### Expected Behavior
Be able to create an Elasticache Cluster with 3 Shards and 2 Replicas per Shard.

### Actual Behavior
Unable to create an Elasticache Cluster with more than 1 shard.  

### Steps to Reproduce
1. There is no argument to specify a cluster mode in "resource aws_elasticache_cluster" 
2. If cluster mode is specified under "aws_elasticache_cluster" resource as an argument "parameter_group_name" with value "default.redis3.2.cluster.on". Then "Cluster mode should be disabled in parameter group" error is thrown.
3. Resource "aws_elasticache_replication_group" does allow to create shard replicas by specifiying "number_cache_cluster" argument; however, there is no argument to specify the number of shards to create. 

### Important Factoids
Neither resource "aws_elasticache_cluster" or "aws_elasticache_replication_group" is capable of creating Elasticache (Redis) cluster with multiple shards and multiple replica per shard. 

### References
None

**Comment by JDiPierro** [comment]
I ran into this today; I tried using `aws_elasticache_cluster` to create a redis cluster. Everything was good but when I was looking in the console I saw the difference between `cluster mode disabled` and `cluster mode enabled`.

After some digging I found that the `aws_elasticache_replication_group` resource has a [`cluster_mode`](https://www.terraform.io/docs/providers/aws/r/elasticache_replication_group.html#cluster_mode) section for creating a `Redis (Cluster mode enabled) cluster`. `num_node_groups` specifies the number of shards and `replicas_per_node_group` specifies the number of replicas per shard.

I feel it'd be helpful if the `aws_elasticache_cluster` resource page mentioned the difference between creating a redis cluster using that resource vs the `aws_elasticache_replication_group`.
